src/rtc.c

```diff
@@ -308,9 +308,15 @@ void RTC_GetTimeGTM(struct gtm *t)
     t->tm_year = _RTC_GetYear(value) - 1900;
 }
 
+
+const char * const RTCNames[] =
+{
+    _tr_noop("Clock"),
+    _tr_noop("Date"),
+};
 const char *RTC_Name(char *str, int i)
 {
-    sprintf(str, "%s", i == 0 ? _tr("Clock") : _tr("Date"));
-    return str;
+    (void)str;
+    return RTCNames[i];
 }
```

Comment:
again, the ternary is safer.  if you don't want to use it, you need to validate that 'i' isn't out of bounds

Reply:
okay